app/lib/state_change_notifier.rb

```diff
@@ -1,85 +1,77 @@
 class StateChangeNotifier
-  def self.sign_up(candidate)
-    candidate_number = Candidate.where(hide_in_reporting: false).count
-
-    return unless (candidate_number % 25).zero?
-
-    candidate_number_is_significant = (candidate_number % 100).zero?
+  APPLICATION_OUTCOME = %i[rejected declined withdrawn recruited].freeze
 
-    human_readable_number = "#{ActiveSupport::NumberHelper.number_to_delimited(candidate_number)}#{candidate_number.ordinal}"
-
-    text = if candidate_number_is_significant
-             ":ultrafastparrot: The #{human_readable_number} candidate just signed up"
-           else
-             ":sparkles: The #{human_readable_number} candidate just signed up"
-           end
-
-    url = helpers.support_interface_candidate_url(candidate)
+  attr_reader :application_choice, :event
 
-    send(text, url)
+  def initialize(event, application_choice)
+    @event = event
+    @application_choice = application_choice
   end
 
-  def self.submit_application(application_form)
-    message = ":rocket: #{application_form.first_name}’s application has been sent to #{application_form.application_choices.map(&:provider).map(&:name).to_sentence}"
-    url = helpers.support_interface_application_form_url(application_form)
-
-    send(message, url)
-  end
+  def application_outcome
+    candidate_name = application_choice.application_form.first_name
+    other_applications = application_choice.self_and_siblings.where.not(id: application_choice.id)
+    providers = [application_choice.course.provider.name]
+
+    if event == :rejected
+      providers += other_applications.select(&:rejected?).map(&:provider).map(&:name)
+
+      declined_message = declined_text_for(other_applications)
+      withdrawn_message = withdrawn_text_for(other_applications)
+      other_applications_outcome = [declined_message, withdrawn_message].compact.to_sentence
```

Comment:
Would it be DRYer to have 2 methods like:

```ruby
def providers_for(event, other_applications)
  return [] if event == :recruited

  other_applications.select(&:"#{event}?") }.map(&:provider).map(&:name)
end

def messages_for(event, other_applications)
  [
    declined_text_for(other_applications),
    rejected_text_for(other_applications),
    withdrawn_text_for(other_applications),
  ].compact
end
```

and use these without the need for the repetition in the conditional clauses. eg.

```ruby
providers += providers_for(event, other_applications)
other_applications_outcome = messages_for(event, other_applications).to_sentence
```

Just a suggestion, not a blocker.

Reply:
It's not as straighforward as that as you don't want, for example, to end up with 'the application has been rejected by all providers.'  and then also include the other rejected provider in the additional outcome message, but I'll take another look as I think there are some improvements that can be made.

Reply:
Ah OK, I see, I wondered if I was missing some context here, it's not a blocker.